src/dive/ui/layout_helper.h

```diff
@@ -0,0 +1,92 @@
+
+/*
+ Copyright 2025 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+#pragma once
+
+#include <QLayout>
+#include <QObject>
+#include <QWidget>
+#include <utility>
+
+#include "absl/base/nullability.h"
+#include "dive/ui/lint.h"
+
+// Examples
+//
+// Setup layout for a widget:
+//   auto layout = NewWidgetLayout<QHBoxLayout>(widget);
+//   auto* label = layout.New<QLabel>(tr("text"));
+//
+// Add a new item to an existing layout:
+//   auto* label = NewWidgetLayout{layout}.New<QLabel>(tr("text"));
```

Comment:
nit: since NewWidgetLayout is a funtion call, should the `{}` be `()` instead?
```suggestion
//   auto* label = NewWidgetLayout(layout).New<QLabel>(tr("text"));
```